libs/@blockprotocol/graph/src/types/subgraph/vertices.ts

```diff
@@ -99,7 +99,7 @@ export const isOntologyTypeVertexId = (
     /** @todo - This means we need to have initialized the type system */
     validateBaseUri(vertexId.baseId).type === "Ok" &&
     "revisionId" in vertexId &&
-    typeof vertexId.revisionId === "number"
+    !Number.isNaN(Number(vertexId.revisionId))
```

Comment:
Is this equivalent?

Reply:
Is it problematic to use `Number` here?
```js
Number(); // 0
Number(null); // 0
Number(""); // 0 
```

Whereas 
```js
parseInt() // NaN
parseInt(null) // NaN
parseInt("") // NaN
```

Although we would then have this..
```js
parseInt("0not a number") // 0
Number("0not a number"); // Nan
```

Reply:
If we want to make sure it's exactly an integer, (without accepting `.0` making it a float) then I think our two options are a regex, or this. 
<img width="499" alt="image" src="https://user-images.githubusercontent.com/25749103/216967228-9aedef14-133d-45cb-a3b0-f6ed81327f9c.png">

Reply:
Seems that this check is better than the regex ([crappy benchmark here](https://jsben.ch/6wCQN))

Reply:
[Introduce and use stringIsNonNegativeInteger](https://github.com/blockprotocol/blockprotocol/pull/939/commits/9dd1b30162bebaa5dc183a01b0fdc4d284cf918d)